File: backend/app/schemas/unified_project.py
"""
Schemas pour le dialogue unifié de création de projet

Ce module permet de créer un projet directement via un dialogue,
sans passer par le formulaire traditionnel.
"""
from datetime import datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field


class StartProjectChatRequest(BaseModel):
    """Requête pour démarrer un dialogue de création de projet"""
    # Pas de champs requis - le dialogue commence vide
    pass


class StartProjectChatResponse(BaseModel):
    """Réponse au démarrage du dialogue"""
    project_id: int
    conversation_id: int  # ID de la conversation d'idéation
    welcome_message: str
    initial_question: str

    model_config = {"from_attributes": True}


class FinalizeProjectRequest(BaseModel):
    """Requête pour finaliser le projet et générer les tâches"""
    project_id: int


class TaskGenerated(BaseModel):
    """Représentation d'une tâche générée"""
    id: int
    title: str
    description: str
    task_type: str
    priority: str
    estimated_duration: Optional[int] = None

    model_config = {"from_attributes": True}


class FinalizeProjectResponse(BaseModel):
    """Réponse à la finalisation du projet"""
    project_id: int
    project_name: str
    project_description: str
    tasks_count: int
    tasks: List[TaskGenerated]
    message: str

    model_config = {"from_attributes": True}

File: frontend/src/services/unifiedProject.ts
/**
 * Service pour la création de projet en mode dialogue unifié
 */
import { api } from './api';
import { STORAGE_KEYS } from '../utils/constants';
import {
  ProjectFinalizationPreview,
  FinalizeValidationRequest,
  ProjectFinalizeResponse
} from '../types/project.types';

export interface StartProjectChatResponse {
  project_id: number;
  conversation_id: number;
  welcome_message: string;
  initial_question: string;
}

export interface TaskGenerated {
  id: number;
  title: string;
  description: string;
  task_type: string;
  priority: string;
  estimated_duration?: number;
}

export interface FinalizeProjectResponse {
  project_id: number;
  project_name: string;
  project_description: string;
  tasks_count: number;
  tasks: TaskGenerated[];
  message: string;
}

class UnifiedProjectService {
  /**
   * Démarre un nouveau projet en mode dialogue
   */
  async startProjectChat(): Promise<StartProjectChatResponse> {
    const token = localStorage.getItem(STORAGE_KEYS.TOKEN);

    const response = await api.post('/projects/start-chat', {}, {
      headers: {
        Authorization: `Bearer ${token}`
      }
    });

    return response.data;
  }

  /**
   * Finalise un projet et génère les tâches automatiquement
   */
  async finalizeProject(projectId: number): Promise<FinalizeProjectResponse> {
    const token = localStorage.getItem(STORAGE_KEYS.TOKEN);

    const response = await api.post('/projects/finalize',
      { project_id: projectId },
      {
        headers: {
          Authorization: `Bearer ${token}`
        }
      }
    );

    return response.data;
  }

  /**
   * Génère un aperçu de finalisation sans persister en base
   * @param projectId - ID du projet en phase IDEATION
   */
  async previewFinalization(projectId: number): Promise<ProjectFinalizationPreview> {
    const token = localStorage.getItem(STORAGE_KEYS.TOKEN);

    const response = await api.post('/projects/preview-finalization',
      { project_id: projectId },
      {
        headers: {
          Authorization: `Bearer ${token}`
        }
      }
    );

    return response.data;
  }

  /**
   * Finalise un projet avec validation utilisateur
   * @param data - Données validées par l'utilisateur (nom, description, tâches approuvées, veille)
   */
  async finalizeWithValidation(data: FinalizeValidationRequest): Promise<ProjectFinalizeResponse> {
    const token = localStorage.getItem(STORAGE_KEYS.TOKEN);

    const response = await api.post('/projects/finalize-with-validation',
      data,
      {
        headers: {
          Authorization: `Bearer ${token}`
        }
      }
    );

    return response.data;
  }
}

export const unifiedProjectService = new UnifiedProjectService();
